### PR TITLE
chore: add an offset tracking example

### PR DESCRIPTION
**Pull request by l4mby**

### CODE REVIEW

File: example/src/offset_tracking_example.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  let lastOffset = initialOffset.value
+  let messageCount = 0
+  const consumerRef = "offset-tracking-consumer"
+  const consumer = await client.declareConsumer(
+    { stream: streamName, offset: initialOffset, consumerRef },
+    (message) => {
+      messageCount++
+      if (message.offset === initialOffset.value) {
+        console.log("First message received")
+      }
```
Comment (gpad) [inline example/src/offset_tracking_example.js:40]: I think you would save `firstOffest` here ...

File: example/src/offset_tracking_example.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  const publisher = await client.declarePublisher({ stream: streamName })
+  const totalMessages = 100
+
+  console.log(`Publishing ${totalMessages} messages`)
+  for (let i = 0; i < totalMessages; i++) {
+    const messageBody = i === totalMessages - 1 ? "marker" : `hello ${i}`
+    await publisher.send(Buffer.from(messageBody))
+  }
+
+  let initialOffset = rabbit.Offset.offset(0n)
```
Comment (gpad) [inline example/src/offset_tracking_example.js:29]:
```suggestion
  const startFrom = rabbit.Offset.offset(0n)
```

File: example/src/offset_tracking_example.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  let firstOffset = initialOffset.value
+  let lastOffset = initialOffset.value
+  let messageCount = 0
+  const consumerRef = "offset-tracking-consumer"
+  const consumer = await client.declareConsumer(
+    { stream: streamName, offset: initialOffset, consumerRef },
```
Comment (gpad) [inline example/src/offset_tracking_example.js:35]:
```suggestion
    { stream: streamName, offset: startFrom, consumerRef },
```

File: example/src/offset_tracking_example.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    (message) => {
+      messageCount++
+      if (message.offset === initialOffset.value) {
```
Comment (gpad) [inline example/src/offset_tracking_example.js:38]:
```suggestion
      if (message.offset === startFrom.value) {
```

File: example/src/offset_tracking_example.js (hunk 1 above)
Comment (dc1992) [reply example/src/offset_tracking_example.js]: This looks resolved

File: example/src/offset_tracking_example.js (hunk 3 above)
Comment (dc1992) [reply example/src/offset_tracking_example.js]: resolved

File: example/src/offset_tracking_example.js (hunk 4 above)
Comment (dc1992) [reply example/src/offset_tracking_example.js]: this has been resolved

**Review by dc1992 — APPROVED** [review]

**Pull request merged**